### ISSUE TITLE
Recents Blacklist

### ISSUE THREAD

**Issue by tnyeanderson** [nextcloud/recommendations#78]
There should be an option to blacklist directories/files (using wildcard notation) from appearing in the "Recents" list at the top of the Files page.

Use case: I have .txt files used with nextcloud notes. I don't want these files to appear in my "recents" list as they are edited often, and edited through the nextcloud/notes app. Because of this, all I see in the recents list are my notes, which then open in text editors rather than the nextcloud/notes app.

I think this is, in addition to not having hidden files a la #71, Will make the experience much better.

**Comment by szaimen** [comment]
Yes, thats really much needed!

**Comment by jonas-arkulpa** [comment]
Have the same need.
My use case: I have one folder which holds all auto uploaded images of my family.
Already excluded it from the gallery (`.nomedia`).

I think it would be useful to reuse this concept here.
For example `.noindex` seems suited:
>Nextant leaves the user the possibility of not indexing the contents of a directory (recursively) by the creation of a .noindex file.

https://nextcloud.com/blog/nextcloud-11-introduces-full-text-search/

**Comment by ChristophWurst** [comment]
> I think it would be useful to reuse this concept here.

Makes sense for power users. But not very usable for the average user. And technically, we have to traverse the files every time we want to build the list of recommendations because that file *might* exist in one of the parent directories. Right now we don't even have to go into the storage to get the information. Storage access will most likely make this feature slow with external storage.

**Comment by Torqu3Wr3nch** [comment]
Came here to open a feature request for this exact feature. I'd settle for just being able to blacklist directories.

**Comment by zzyzx-dc** [comment]
Seconding this as well. I use the auto-upload feature of Nextcloud and all my photo uploads dominated the entire Recommended / Most Recent list.

Now I'm using Joplin which works well with Nextcloud. But every note edit modifies a file within Nextcloud directory `/Joplin`, so my Recommended / Most Recent files are just an endless list of hashed .md files.

**Comment by Thesola10** [comment]
I think the "flag" to exclude a folder from Recents should be a hidden file or WebDAV attribute, such that applications which use Nextcloud as storage (like Swift Backup or Seedvault) could automatically set the flag without the user's intervention.

**Comment by indyspike** [comment]
The bookmarks app is now creating backups and the recommended files is  flooded with YYYY-MM-DD.html files. Either a mechanism of blacklisting directories, or the ability to select which directories are to be looked at for entries for the recommended list would be a good enhancement.

**Comment by gmeks** [comment]
Perhaps it should follow the .noindex or .nomedia files? 

This will also allow you to easy filter away backups, and means that 1 "setting" works consistent accross many apps.

**Comment by tnyeanderson** [comment]
I definitely prefer the flexibility of a separate option. I might want to make sure a folder is indexed, but not showed in recents. Or show up in my photos app but not in recents.

`.norecents` ?

**Comment by j0hnnyclaymore** [comment]
+1 Joplin syncs to my nextcloud and dashboard recommended files is now useless.

**Comment by xinstein** [comment]
> +1 Joplin syncs to my nextcloud and dashboard recommended files is now useless.

+1 That's exactly when I disabled the recommended files card.

**Comment by ChristophWurst** [comment]
If .nomedia works well for photos we can add it here as well, I guess. I'm by no means an expert with file operations and can't judge the performance concerns.

A setting for .nomedia, .noimage, .norecents etc seems a bit too much. Nextcloud design tries to find good defaults and limit the number of settings a user needs.

**Comment by mufeedali** [comment]
> Nextcloud design tries to find good defaults and limit the number of settings a user needs.

I think the point of ".norecents" should be that the application automatically adds it. For example, the Notes app in the Notes folder or Bookmark apps in the backup folder. Also, what if the user has a folder inside a folder with photos that he doesn't want the media to be scanned for but wants them to show up in recents?

**Comment by kds69** [comment]
up

**Comment by Ngineer4life** [comment]
Agree, need this functionality too. Would see it as a setting where you can choose directories, and the app adds .norecents, or does the necessary plumbing. Don’t is user friendly and encourages users to use Recommended Files

**Comment by henrysheehy** [comment]
Bump

**Comment by ihatemyisp** [comment]
Like many others, I believe this needs to be implemented. I'll throw my hand in to say .norecents is the better option.

**Comment by pretubedongit** [comment]
Exactly the same situation. Syncing Joplin with nextcloud. Only md files are in the recent list.

**Comment by k1n6b0b** [comment]
dot files appear to be the Defacto way to exclude folders from nextcloud apps - here's another vote for '.norecents' 

My recent files shows the same list of external_files (movies in my case) and never changes. I'll disable the app for now which seems to be the resolution on the internet, but i'd love to have insights into my interesting files!

**Comment by jm355** [comment]
Even just hiding hidden files from recents would be a great start. Seedvault via grapheneos uses .SeedVaultAndroidBackup and it fills the recommended files for me

**Comment by MHanak1** [comment]
I have also noticed that nextcloud includes content from hidden folders in the recents view. I have a hidden folder called .apps, and within it i keep things like my Joplin notes. Despite it not being accessible through the folders view, the encrypted .md files still show up in recents.

I think on top of adding a .norecents/.noindex marker hidden folders should also be excluded

**Comment by MHanak1** [comment]
I have also noticed that nextcloud includes content from hidden folders in the recents view. I have a hidden folder called .apps, and within it i keep things like my Joplin notes. Despite it not being accessible through the folders view, the encrypted .md files still show up in recents.

I think on top of adding a .norecents/.noindex marker hidden folders should also be excluded
